File: week01/src/sum3.ts
import { createInterface } from 'readline';

// initializing the input, output interface
const readline = createInterface({
  input: process.stdin,
  output: process.stdout,
});

// reading a line form the cli and converting it to a number
const readLineNumberAsync: (msg: string) => Promise<number> = (msg: string) => {
  return new Promise((resolve) => {
    readline.question(msg, (userRes: string) => {
      resolve(Number(userRes));
    });
  });
};

// a function to calculate the sum of three numbers
function sum(number1: number, number2: number, number3: number): number {
  return number1 + number2 + number3;
}

async function run() {
  const message = 'Enter a number: ';
  const number1 = await readLineNumberAsync(message);
  const number2 = await readLineNumberAsync(message);
  const number3 = await readLineNumberAsync(message);

  console.log('The sum is: ' + sum(number1, number2, number3));
  readline.close();
}

run();

File: week01/src/sum3.cpp
#include <iostream>

using namespace std;

int sum(int number1, int number2, int number3);

int main() {

    // the input values for the sum function
    double number1, number2, number3;

    cout << "Enter three numbers separated by spaces " << endl;
    cin >> number1 >> number2 >> number3;
    cout << "The sum between the numbers is: " << sum(number1, number2, number3) << endl;

    return 0;
}


int sum(int number1, int number2, int number3) {
    return number1 + number2 + number3;
}


